File: Backend/Tafe/Models/Ingredient.cs
#nullable disable
using Tafe.Models;

public class Ingredient : EntityTemplate
{
    public decimal Quantity { get; set; }      // الموجود بالمخزن

    public int UnitId { get; set; }
    public Unit Unit { get; set; }

    public decimal MinQuantityAlert { get; set; }

    public ICollection<ProductIngredient> Products { get; set; } = [];
    public virtual ICollection<StockTransaction> StockTransactions { get; set; } = [];

    public virtual ICollection<PurchaseInvoiceItem> PurchaseInvoiceItems { get; set; } = [];
}

File: Frontend/src/types/ingredient.ts
import type { Unit } from "./unit";

export interface Ingredient {
  id: number;
  name: string;
  minQuantityAlert: number;
  unit: Unit;
  quantity: number;
}

export interface IngredientCreate {
  name: string;
  minQuantityAlert: number;
  unitId: number;
}

export interface IngredientUpdate {
  id: number;
  name: string;
  minQuantityAlert: number;
  unitId: number;
}

export interface IngredientWarning {
  id: number;
  name: string;
  quantity: number;
  minQuantityAlert: number;
  unit: string;
}